### ISSUE TITLE
Easybuild in container singularity 2.4.2 works fine and singularity 3.5.2 is not working

### ISSUE THREAD

**Issue by smarocchi** [easybuilders/easybuild-framework#3232]
Dears, 
           I played a little bit with the container recipes made by easybuild 4.1.1 (I know it is experimental but..). I noticed 2 main issues:

1) With the last versions of easybuild> 4.0.0 the lines 

pip install 'vsc-install<0.11.4' 'vsc-base<2.9.0'

it is not necessary anymore and it raise an error.

2) With recent version of singularity (e.g. 3.5.X ) the commands 

```
# change to 'easybuild' user
su - easybuild
```

does not go into a normal user inside the container and  Easybuild raise an error because it is convinced to install software as root

instead with singularity 2.4.2 everything works properly. 

How is it possible to solve this ?

Why singularity newer versions >2.4.2 are not available inside easybuild stack software ?  

If I build an image with Singularity 2.4.2 is it still possible to run it with newer versions of singularity ?

Thanks.

**Comment by sassy-crick** [comment]
@boegel 

A potential solution for this problem:
```
# change to 'easybuild' user
su - easybuild
```

might be what I am doing:

```
# configure EasyBuild
cat > /tmp/eb-install.sh << 'EOF'
#!/bin/bash  
echo $USER 
export EASYBUILD_PREFIX=/scratch 
export EASYBUILD_TMPDIR=/scratch/tmp 
export EASYBUILD_SOURCEPATH=/scratch/sources:/tmp/easybuild/sources 
export EASYBUILD_INSTALLPATH=/app 
eb tbb-2019_U4-GCCcore-8.2.0.eb --robot 
mkdir -p /app/lmodcache 
$LMOD_DIR/update_lmod_system_cache_files -d /app/lmodcache -t /app/lmodcache/timestamp /app/modules/all  
EOF

chmod a+x /tmp/eb-install.sh

su -l easybuild -c /tmp/eb-install.sh
```
This way, it is executed as 'easybuild' and not as root. I guess that should work in older versions of Singularity as well and I hope that is also quite portable to other version. 

I hope it helps!

**Comment by boegel** [comment]
This problem is fixed with the change in #3347, as proposed by @sassy-crick here and @shahzebsiddiqui in #3171
The fix will be included in the next EasyBuild release (4.2.2 or 4.3.0, not determined yet).
